packages/reactive-dict/reactive-dict.js

```diff
@@ -1,117 +1,114 @@
 // XXX come up with a serialization method which canonicalizes object key
 // order, which would allow us to use objects as values for equals.
-var stringify = function (value) {
-  if (value === undefined)
+function stringify(value) {
+  if (value === undefined) {
     return 'undefined';
+  }
   return EJSON.stringify(value);
-};
-var parse = function (serialized) {
-  if (serialized === undefined || serialized === 'undefined')
+}
+
+function parse(serialized) {
+  if (serialized === undefined || serialized === 'undefined') {
     return undefined;
+  }
   return EJSON.parse(serialized);
-};
+}
 
-var changed = function (v) {
+function changed(v) {
   v && v.changed();
-};
+}
 
 // XXX COMPAT WITH 0.9.1 : accept migrationData instead of dictName
-ReactiveDict = function (dictName) {
-  // this.keys: key -> value
-  if (dictName) {
-    if (typeof dictName === 'string') {
-      // the normal case, argument is a string name.
-      // _registerDictForMigrate will throw an error on duplicate name.
-      ReactiveDict._registerDictForMigrate(dictName, this);
-      this.keys = ReactiveDict._loadMigratedDict(dictName) || {};
-      this.name = dictName;
-    } else if (typeof dictName === 'object') {
-      // back-compat case: dictName is actually migrationData
-      this.keys = dictName;
-    } else {
-      throw new Error("Invalid ReactiveDict argument: " + dictName);
-    }
-  } else {
-    // no name given; no migration will be performed
+export class ReactiveDict {
+  constructor(dictName, dictData = {}) {
+    // this.keys: key -> value
     this.keys = {};
-  }
 
-  this.allDeps = new Tracker.Dependency;
-  this.keyDeps = {}; // key -> Dependency
-  this.keyValueDeps = {}; // key -> Dependency
-};
+    if (dictName) {
+      // name given; migration will be performed
+      if (typeof dictName === 'string') {
+        // the normal case, argument is a string name.
+        // _registerDictForMigrate will throw an error on duplicate name.
+        ReactiveDict._registerDictForMigrate(dictName, this);
+        // Use _setObject to make sure values are stringified
+        this._setObject(ReactiveDict._loadMigratedDict(dictName) || dictData);
+        this.name = dictName;
+      } else if (typeof dictName === 'object') {
+        // back-compat case: dictName is actually migrationData
+        // Use _setObject to make sure values are stringified
+        this._setObject(dictName);
```

Comment:
I actually had guards here but somewhere they got lost before I created the PR. I'll fix this. I wanted the dependencies to be undefined here when this function is run from constructor so that no unnecessary invalidations are triggered when we set a bunch of initial values.